pkg/skaffold/schema/profiles.go

```diff
@@ -80,45 +90,78 @@ func checkKubeContextConsistency(contextSpecificProfiles []string, cliContext, e
 
 // activatedProfiles returns the activated profiles and activated profiles which are kube-context specific.
 // The latter matters for error reporting when the effective kube-context changes.
-func activatedProfiles(profiles []latest.Profile, opts cfg.SkaffoldOptions) ([]string, []string, error) {
-	var activated []string
-	var contextSpecificProfiles []string
+func activatedProfiles(profiles reflect.Value, version semver.Version, opts cfg.SkaffoldOptions) ([]string, []string, error) {
+	activated, contextSpecificProfiles, err := checkActivations(profiles, version, opts)
+	if err != nil {
+		return nil, nil, err
+	}
 
-	if opts.ProfileAutoActivation {
-		// Auto-activated profiles
-		for _, profile := range profiles {
-			for _, cond := range profile.Activation {
-				command := isCommand(cond.Command, opts)
+	for _, profile := range opts.Profiles {
+		if strings.HasPrefix(profile, "-") {
+			activated = removeValue(activated, strings.TrimPrefix(profile, "-"))
+		} else {
+			activated = append(activated, profile)
+		}
+	}
 
-				env, err := isEnv(cond.Env)
-				if err != nil {
-					return nil, nil, err
-				}
+	return activated, contextSpecificProfiles, nil
+}
+
+// checkActivations converts each profile `Activation` to a known version and applies the corresponding `checkActivation` function.
+func checkActivations(profiles reflect.Value, version semver.Version, opts cfg.SkaffoldOptions) ([]string, []string, error) {
+	if !opts.ProfileAutoActivation {
+		return nil, nil, nil
+	}
 
-				kubeContext, err := isKubeContext(cond.KubeContext, opts)
+	var activatedProfiles []string
+	var contextSpecificProfiles []string
+	v1b4, _ := apiversion.Parse(v1beta4.Version)
```

Comment:
Can we extract this (and the later use) to a constant?  (Will need to add `apiversion.MustParse()`?)